### PR TITLE
Python 3 migration of tests - io/disk/

### PR DESCRIPTION
**Pull request by narasimhan-v**
Python 3 migration of tests - io/disk/

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
```
JOB ID     : 5452a13001b9f83fcf51cc1ba542105b852c21f8
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.04-5452a13/job.log
 (1/1) bonnie.py:Bonnie.test;dir-ext4-c865: PASS (5.78 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 5.93 s
JOB ID     : 9de1f7526a6f2d21ec5e623c16824f39adffa20a
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.04-9de1f75/job.log
 (1/2) disk_info.py:DiskInfo.test_commands;dir-xfs-1e71: PASS (12.80 s)
 (2/2) disk_info.py:DiskInfo.test;dir-xfs-1e71: ERROR: [Errno 2] No such file or directory: '/sys/block/mpathb/queue/physical_block_size' (7.46 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 20.46 s
JOB ID     : 279c8b1516a8f697f57bc8863c35451a3203d22f
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.05-279c8b1/job.log
 (1/2) disktest.py:Disktest.test;dir-xfs-1e71: CANCEL: Chunk size has to be greater or equal to RAM size. (38000 > 39860) (0.55 s)
 (2/2) disktest.py:Disktest.test;dir-btrfs-405f: CANCEL: Chunk size has to be greater or equal to RAM size. (38000 > 39860) (0.56 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB TIME   : 1.31 s
JOB ID     : 3921a476caf242b6121777de3ef5a534cf5a0d04
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.05-3921a47/job.log
 (1/3) htx_block_devices.py:HtxTest.test_start;run-3dfd: PASS (73.29 s)
 (2/3) htx_block_devices.py:HtxTest.test_check;run-3dfd: PASS (60.11 s)
 (3/3) htx_block_devices.py:HtxTest.test_stop;run-3dfd: PASS (11.27 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 144.94 s
JOB ID     : 3cf404a2be620e7827c2c554fa4e27dfbf380bc1
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.07-3cf404a/job.log
 (1/1) ltp_fs.py:LtpFs.test_fs_run;fs_di-dir-xfs-57d7: PASS (163.25 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 163.40 s
JOB ID     : 97f42dc9a3dfe2afef92d7f16ea8effd6e61afec
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.10-97f42dc/job.log
 (1/3) lvsetup.py:Lvsetup.test;xfs-lv_size-5c37: PASS (4.37 s)
 (2/3) lvsetup.py:Lvsetup.test_vg_recreate;xfs-lv_size-5c37: PASS (45.34 s)
 (3/3) lvsetup.py:Lvsetup.test_lv_snapshot;xfs-lv_size-5c37: PASS (8.17 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 58.08 s
JOB ID     : 4b64ec73e1c2ddd0dff2172ed019a8b40eee7be2
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.11-4b64ec7/job.log
 (1/3) multipath_test.py:MultipathTest.test;queue-length-c9e4: PASS (585.92 s)
 (2/3) multipath_test.py:MultipathTest.test;service-time-3433: PASS (592.72 s)
 (3/3) multipath_test.py:MultipathTest.test;round-robin-d930: PASS (592.01 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1770.90 s
JOB ID     : 88ccab913b5347c45db8fc2cd07ff4ba617010c4
JOB LOG    : /root/avocado/job-results/job-2019-07-30T12.41-88ccab9/job.log
 (1/1) tiobench.py:Tiobench.test;8192-1024-dir-xfs-2-10-fe26: PASS (29.29 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 29.45 s
JOB ID     : ca28a20cb464ef9296b564452d60218e80d72ad0
JOB LOG    : /root/avocado/job-results/job-2019-07-30T11.05-ca28a20/job.log
 (1/4) softwareraid.py:SoftwareRaid.test_run;superblock-1.2-raid1-79d9: PASS (2289.26 s)
 (2/4) softwareraid.py:SoftwareRaid.test_run;superblock-1.2-raid5-a147: PASS (491.61 s)
 (3/4) softwareraid.py:SoftwareRaid.test_run;superblock-1.2-raid10-2290: PASS (246.72 s)
 (4/4) softwareraid.py:SoftwareRaid.test_run;superblock-1.2-raid6-7d65: FAIL: Failed to create a MD device (0.61 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 3028.52 s
```

File: io/disk/disk_info.py [hunk 1]
```diff
@@ -130,9 +129,11 @@ def test(self):
         # Get the size and UUID of the disk
         cmd = "lsblk -l %s --output SIZE -b |sed -n 2p" % self.disk
         output = process.system_output(cmd, ignore_status=True,
-                                       shell=True, sudo=True)
-        self.size_bytes = (output.strip("\n"))[0]
-        self.log.info("Disk: %s Size: %s", self.disk, self.size_bytes)
+                                       shell=True, sudo=True).decode("utf-8")
+        if not output:
+            self.cancel("No information available in lsblk")
+        self.size_b = (output.strip("\n"))
```
Comment (harish-24) [inline io/disk/disk_info.py]: As output is already decoded as string, why is `[0]` element removed here. Is it a conscious change?

File: io/disk/disk_info.py [hunk 2]
```diff
@@ -151,25 +152,27 @@ def test(self):
...
                                        ignore_status=True, shell=True,
                                        sudo=True)
-        if sector_string not in output:
+        if sector_string not in output.decode("utf-8"):
             msg.append("Mismatch in sector sizes of lbs,pbs in "
                        "fdisk o/p w.r.t sysfs paths")
         io_size_string = "I/O size (minimum/optimal): %s " \
                          "bytes / %s bytes" % (mis, mis)
-        if io_size_string not in output:
+        if io_size_string not in output.decode("utf-8"):
```
Comment (harish-24) [inline io/disk/disk_info.py]: Better to decode output at `process.system_output` line if it is not used for other purpose.

File: io/disk/disk_info.py [hunk 3]
```diff
@@ -191,20 +194,20 @@ def test(self):
...
-        self.uuid = output.split('"')[1]
+        self.uuid = output.decode("utf-8").split('"')[1]
         self.log.info("Disk: %s UUID: %s", self.disk, self.uuid)
 
         # Verify mount point, filesystem type and UUID for each test variant
         output = process.system_output("lsblk -l %s" % self.disk,
                                        ignore_status=True, shell=True,
                                        sudo=True)
-        if self.dirs in output:
+        if self.dirs in output.decode("utf-8"):
```
Comment (harish-24) [inline io/disk/disk_info.py]: As decode is used in many places at `system_output` line, for uniformity better to maintain at same way through out the file.

File: io/disk/htx_block_devices.py [hunk 4]
```diff
@@ -46,7 +46,8 @@ def setUp(self):
         """
         Setup
         """
-        if 'ppc64' not in process.system_output('uname -a', shell=True):
+        if 'ppc64' not in process.system_output('uname -a',
```
Comment (harish-24) [inline io/disk/htx_block_devices.py]: As you are already handling, `distro.detect().arch` can be used which also avoids decoding here.

File: io/disk/disk_info.py (hunk 1 above)
Comment (narasimhan-v) [reply io/disk/disk_info.py]: Thanks for the catch. Was debugging, and missed it.

File: io/disk/disk_info.py [hunk 5]
```diff
@@ -190,21 +193,21 @@ def test(self):
...
-                                       sudo=True)
+                                       sudo=True).decode("utf-8")
         if self.dirs in output:
             self.log.info("Mount point %s for disk %s updated in lsblk o/p",
                           self.dirs, self.disk)
         output = process.system_output("df %s" % self.disk,
                                        ignore_status=True, shell=True,
                                        sudo=True)
-        if self.dirs in output:
+        if self.dirs in output.decode("utf-8"):
```
Comment (harish-24) [inline io/disk/disk_info.py]: This is OK, would it be better to move this also to `process.system_output`?

**Pull request merged**